modules/auxiliary/gather/asrep.rb

```diff
@@ -77,26 +77,36 @@ def run
   def run_brute
     result_count = 0
     user_file = datastore['USER_FILE']
-    if user_file.nil?
-      fail_with(Msf::Module::Failure::BadConfig, 'User file must be specified when brute forcing')
+    username = datastore['USERNAME']
+    if user_file.blank? && username.blank?
+      fail_with(Msf::Module::Failure::BadConfig, 'User file or username must be specified when brute forcing')
+    end
+    if username.present?
```

Comment:
Paper trail: I was aligning this module to more closely align with how login scanners work, where you can provide a user file, or a username.

This might be confusing for the auth scenario to LDAP - we can revert this change if so :+1: